cmake/modules/rust.cmake

```diff
@@ -142,14 +168,19 @@ ${config_paths}
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   )
 
+  # Be sure we don't try building this until al of the generated headers have been generated.
   add_custom_target(librustapp ALL
     DEPENDS ${DUMMY_FILE}
+        # The variables, defined at the top level, don't seem to be accessible here.
```

Comment:
Does anyone know how to do this right? I'd like to use the `${SYSCALL_LIST_H_TARGET}` instead of its expansion, but that variable doesn't seem to be defined at this point in this module.

Reply:
`rust.cmake` is executed first, defining the `rust_cargo_application()` function. At this point `SYSCALL_LIST_H_TARGET` is not defined yet:
![image](https://github.com/user-attachments/assets/e753772d-08d0-4fe0-9ce0-50e0e1ee2f27)

Then comes the Zephyr `CMakeLists.txt` defining `SYSCALL_LIST_H_TARGET`:
![image](https://github.com/user-attachments/assets/5ed77c5d-4c88-4b58-9068-b2ea51702f82)

Finally the `rust_cargo_application()` function is executed, but the variables from Zephyr `CMakeLists.txt` are alreadyout of scope as can be seen with the stack trace:
![image](https://github.com/user-attachments/assets/0be24cd5-cc4d-47cd-8ad5-6a90663a0fa8)

I don't see how this could be done without modifying something elsewhere. Would it maybe be reasonable to just add `zephyr_interface` as a dependency?

Reply:
I'll try adding `zephyr_interface` as a dependency, and see if it works. It's a little frustrating because the failure are somewhat non-deterministic, although it does seem to fail pretty consistently in CI.

Reply:
So `zephyr_interface` doesn't really work as a dependency, as it seems to just hold the flags needed to set include paths for the zephyr interfaces. It doesn't actually have any header files in it. It seems that the generated targest are what is needed to have the dependencies work.

Reply:
Have you tried with `add_dependencies(librustapp zephyr_interface)`? I think the `DEPENDS` parameter only works for files, not for targets.